### PR TITLE
Add new hashes introduced in Python 3.6.

### PR DESCRIPTION
**Pull request by jelford**
The new functions are documented [here](https://docs.python.org/3.6/library/hashlib.html?highlight=hashlib#hash-algorithms). `VarLenHash` is just a copy of `Hash` with the relevant signatures changed to take into account the `length` argument.

* sha3 family of hash functions
* shake variable length hashes that were introduced alongside sha3
* blake2 family of hash functions

### CODE REVIEW

File: stdlib/3/hashlib.pyi [hunk 1]
```diff
@@ -40,3 +40,39 @@ algorithms_available = ...  # type: AbstractSet[str]
 # New in version 3.4
 if sys.version_info >= (3, 4):
     def pbkdf2_hmac(hash_name: str, password: _DataType, salt: _DataType, iterations: int, dklen: Optional[int] = ...) -> bytes: ...
+
+if sys.version_info >= (3, 6):
+    class VarLenHash(metaclass=ABCMeta):
```
Comment (JelleZijlstra) [inline stdlib/3/hashlib.pyi]: This class doesn't actually exist as `hashlib.VarLenHash`, so it should be marked as private (`_VarLenHash`).

Looks like the same is true for Hash, but that's not new to this PR.

Comment (jelford) [inline stdlib/3/hashlib.pyi]: Pushed; is it worth doing the clean-up on `Hash` at the same time?

Comment (JelleZijlstra) [inline stdlib/3/hashlib.pyi]: Yes please.

**Comment by jelford** [comment]
Err, sorry, got a bit trigger happy there. I'll come back to this tomorrow night (UK) when I should be able to look at it properly.

**Comment by jelford** [comment]
Hi @JelleZijlstra, sorry it took a bit longer than expected, but I think the changes I've just pushed up should be reasonable now. I made `Hash` private, as discussed.

**Comment by JelleZijlstra** [comment]
Thanks! All good now.

**Pull request merged**